update Overview/Artifact (#1048)
mendersoftware/mender-docs: 02.Overview/03.Artifact/docs.md

    category: docs
---

In order to ensure a secure and robust update process, Mender needs *additional metadata*
alongside the raw bits of the update payload. Depending on the version of the Artifact used,
the metadata might be different, but must contain:

* *Name* of the software build, used to track installed software on a given device
  and ensures that identical software versions are not redeployed.
* *Device types* the software is *compatible* with, so that software is not
  deployed to incompatible hardware (e.g. CPU architecture, hardware floating
  point support, peripheral drivers).

> * *Checksum* of the payload, so that software is not run if it gets corrupted

Author: Ok to update with to:

```
but must contain the:

* *Name*...
* *Device types*...
* *Checksum*...
```

?

> * *Checksum* of the payload, so that software is not run if it gets corrupted

Reviewer: Yes, better imo :+1:

  during transit or storage.

## The Mender Artifact file format

To handle the requirements mentioned above, Mender defines and uses a
specific file format, identified by its `.mender` suffix which is also known as

> a **Mender Artifact** or simply **Artifact**. All relevant components of Mender,

Author: Applied locally.

> such as the client and server, understand and use (only) this specific file

Reviewer: ```suggestion
such as the client and server, understand and use only this specific file
```
Don't think the parenthesis add much here.

> format when working with software deployments.

Reviewer: ```suggestion
format when doing software deployments.
```

Internally, a Mender Artifact is simply a file archive in the [tarball format](https://en.wikipedia.org/wiki/Tar_(computing)?target=_blank). It contains several files for incorporating

> versioning, extensions and metadata, as well as the payload file(s).

Reviewer: ```suggestion
versioning, extensions, metadata and the payload file(s).
```

> The payload within Artifacts is compressed by default and there are several

Reviewer: ```suggestion
Artifact payloads are compressed by default and there are several
```

> The payload within Artifacts is compressed by default and there are several

Reviewer: ```suggestion
The payload within Artifacts is compressed by default and supports multiple compression algorithms
```

> The payload within Artifacts is compressed by default and there are several

Reviewer: Maybe even:
```suggestion
Artifact payloads support multiple compression algorithms, and is compressed by default.
```

compression algorithms supported.

The diagram below shows an example of the main attributes and structure of a

> Mender Artifact file. Please note that the exact format of the artifact may vary

Reviewer: ```suggestion

! Note that the exact format of the artifact may vary
```

> Mender Artifact file. Please note that the exact format of the artifact may vary

Reviewer: ```suggestion
Mender Artifact file. Please note that the exact format of the Artifact may vary
```

between versions.

![Mender Artifact format](mender-artifact-format.png)

<!--AUTOVERSION: "mender-artifact/blob/%"/mender-artifact-->

> You can find more details about the Mender Artifact format at

Reviewer: ```suggestion
You can find more details about the Mender Artifact format in the
```

[Mender Artifact file documentation](https://github.com/mendersoftware/mender-artifact/blob/master/Documentation?target=_blank).

## Versions

Mender is constantly evolving to adapt to the needs of its users, and the Mender
Artifact format has several revisions. See the
[compatibility documentation](../14.Compatibility/docs.md) for an overview of

> which versions of the Mender Artifact format are supported by Mender clients.

Reviewer: ```suggestion
which versions of the Mender Artifact format are supported by which Mender client versions.
```

> ## Working with Mender Artifacts

Reviewer: ```suggestion
## Work with Mender Artifacts
```

The command-line utility `mender-artifact` is the best way to work directly with
Mender Artifacts. It allows you to view, create, modify and sign all types of

> Mender Artifacts. Get it at [the Downloads section](../../08.Downloads/docs.md).

Reviewer: ```suggestion
Mender Artifacts. Get it at [the Downloads section](../../08.Downloads/docs.md#mender-artifact).
```

> Mender Artifacts. Get it at [the Downloads section](../../08.Downloads/docs.md).

Reviewer: ```suggestion
Mender Artifacts. Get it in the [the Downloads section](../../08.Downloads/docs.md).
```
I'm not sure at is the right preposition here. None native english speaker though, so... Who knows really

Mender also has OS-level integrations for creating system updates as part of
your existing OS build process. Learn more in the

> [Creating Artifacts](../../04.Artifacts/chapter.md) section.

Reviewer: ```suggestion
[Creating Artifacts section](../../04.Artifacts/chapter.md).
```
Then this will mirror the link description above :)

Author: What do you mean?

Reviewer: It is nothing really, but:
Above:
```
Get it in the [Downloads section]
```
Here:
```
[Creating Artifact](foobar) section
```

Also, maybe it should be:

`Create Artifacts`, while we're at it?

Author: Got it, aligned.

Unfortunately the target section is called "Creating Artifacts", so for now I think it would leave it as is.

> ### Signing and verification

Reviewer: ```suggestion
### Sign and verify
```
More active, no?

To verify that the Artifact comes from a trusted source, the Mender Artifact
format supports end-to-end image signing and verification. In order to create a

> signed Artifact, please follow the instructions at

Reviewer: ```suggestion
signed Artifact, please follow the instructions in the
```

> [Signing and verifying Mender Artifact](../../04.Artifacts/40.Signing-and-verification/docs.md).

Reviewer: ```suggestion
[Signing and verifying a Mender Artifact section](../../04.Artifacts/40.Signing-and-verification/docs.md).
```
